src/main/scala/linting/LintAnonymousRegisters.scala

```diff
@@ -0,0 +1,134 @@
+package freechips.rocketchip.linting
+
+import firrtl._
+import firrtl.ir._
+import firrtl.traversals.Foreachers._
+import firrtl.annotations.NoTargetAnnotation
+import firrtl.options.{OptionsException, RegisteredLibrary, ShellOption}
+import firrtl.stage.RunFirrtlTransformAnnotation
+import chisel3.experimental.ChiselAnnotation
+
+
+import scala.collection.mutable
+
+case class LintError(info: FileInfo, modules: Set[String])
+case class LintExceptions(seq: Seq[LintError]) extends FirrtlUserException(
+  s"""Linting Report: ${seq.size} exceptions!
+     |Please fix using @chiselName, *.suggestName(...). Otherwise, whitelist the file via the Chisel commandline option:
+     |    --linting-whitelist <filename>.scala
+     |or via annotations in your Chisel code:
+     |    annotate(LintAnonymousRegsWhitelist("<filename>.scala")).
+     |
+     |Only showing first few exceptions:
+     |${seq.zip(0 until 20)
+           .map { case (lint: LintError, idx: Int) => s"$idx. ${lint.info} in ${lint.modules}"}
+           .mkString("\n")}
+     """.stripMargin
```

Comment:
Why only show the first 20? If we trim by default we should provide an option to see them all.

Reply:
if we only show the first 20 or either way, can you make it obvious how to see the reset? Something like Only showing the first few execptions, see ${filepath} for full report"?